File: src/Tawtheef.Websites/operations-web/src/app/pages/user/employee/test-slots-management/models/test-slot-list-item.dto.ts
import { dropdownOptionsModel } from "../../../../../shared/models/dropdown-options.model";

export interface TestSlotListItemDto {
  id: string;
  title: string;
  roomId: string;
  roomName: string;
  slotDate: string;
  startTime: string;
  endTime: string;
  candidateCount: number;
  hallSupervisorId?: string;
  hallSupervisorName?: string;
  status: dropdownOptionsModel;
}

export interface TestSlotFilters {
  pageNumber: number;
  pageSize: number;
  language: string;
  searchTerm?: string;
  roomId?: string;
  dateFrom?: string;
  dateTo?: string;
}

File: src/Application.Operation/Features/Employee/TestSlots/DTOs/TestSlotListItemDto.cs
using Tawtheef.Application.Common.Models;

namespace Application.Operation.Features.Employee.TestSlots.DTOs;

public sealed record TestSlotListItemDto
{
    public Guid Id { get; init; }
    public required string Title { get; init; }
    public Guid RoomId { get; init; }
    public required string RoomName { get; init; }
    public DateOnly SlotDate { get; init; }
    public TimeOnly StartTime { get; init; }
    public TimeOnly EndTime { get; init; }
    public int CandidateCount { get; init; }
    public Guid? HallSupervisorId { get; init; }
    public string? HallSupervisorName { get; init; }
    public bool IsCurrentUserAssigned { get; init; }
    public bool IsCurrentUserRoomHead { get; init; }
    public required DropdownOptions Status { get; init; }
}
